onnxruntime/core/optimizer/nhwc_transformer.cc

```diff
@@ -213,6 +214,23 @@ void NhwcTransformerImpl::TransformQLinearActivation(Node& node) {
   CreateNhwcArgument(node, node, nhwc_input->rank_);
 }
 
+void NhwcTransformerImpl::TransformQLinearGlobalAveragePool(Node& node) {
+  auto& input_defs = node.MutableInputDefs();
+
```

Comment:
You should add a check that channels_last is not already 1.

Reply:
Add check with ORT_ENFORCE.